executor/diagnostics_test.go

```diff
@@ -166,17 +169,136 @@ func (s *diagnosticsSuite) TestInspectionResult(c *C) {
 	}
 }
 
+func (s *diagnosticsSuite) parseTime(c *C, se session.Session, str string) types.Time {
+	t, err := types.ParseTime(se.GetSessionVars().StmtCtx, str, mysql.TypeDatetime, types.MaxFsp)
+	c.Assert(err, IsNil)
+	return t
+}
+
+func (s *diagnosticsSuite) TestThresholdCheckInspection(c *C) {
+	configurations := []string{
+		"tikv,127.0.0.1,raftstore.apply-pool-size,2",
+		"tikv,127.0.0.1,raftstore.store-pool-size,2",
+		"tikv,127.0.0.1,readpool.coprocessor.high-concurrency,4",
+		"tikv,127.0.0.1,readpool.coprocessor.low-concurrency,4",
+		"tikv,127.0.0.1,readpool.coprocessor.normal-concurrency,4",
+		"tikv,127.0.0.1,readpool.storage.high-concurrency,4",
+		"tikv,127.0.0.1,readpool.storage.low-concurrency,4",
+		"tikv,127.0.0.1,readpool.storage.normal-concurrency,4",
+		"tikv,127.0.0.1,server.grpc-concurrency,8",
+		"tikv,127.0.0.1,storage.scheduler-worker-pool-size,6",
+	}
+
+	fpName := "github.com/pingcap/tidb/infoschema/mockInspectionSchemaClusterConfigData"
+	fpExpr := strings.Join(configurations, ";")
+	c.Assert(failpoint.Enable(fpName, fmt.Sprintf(`return("%s")`, fpExpr)), IsNil)
```

Comment:
It's better to reuse the exists failpoint (`mockMergeMockInspectionTables`) to mock data.